### ISSUE TITLE
Cannot configure new profiles using aws configure --profile option in v1.25.54

### ISSUE THREAD

**Issue by kbehlers** [aws/aws-cli#7199]
### Describe the bug

In v1.25.54
```
aws configure --profile a_new_profile
```
now fails with
```
The config profile (a_new_profile) could not be found
```
unless the profile already exists in ~/.aws/credentials.

### Expected Behavior

The expected behavior is in v1.25.53 where
```
aws configure --profile a_new_profile
```
starts the interactive creation prompts.

### Current Behavior

In v1.25.54
```
aws configure --profile a_new_profile
```
now fails with
```
The config profile (a_new_profile) could not be found
```
unless the profile already exists in ~/.aws/credentials.

### Reproduction Steps

1. pip install awscli==1.25.54 in a new virtual environment
2. try to create a new profile (one that doesn't already exist in ~/.aws/credentials)

```
aws configure --profile a_new_profile
```

### Possible Solution

Based on the diff https://github.com/aws/aws-cli/compare/1.25.53...1.25.54

The changes to the command parsing in https://github.com/aws/aws-cli/pull/7114 seem like the most likely source of the problem.

### CLI version used

1.25.54

### Environment details (OS name and version, etc.)

macOS Monterey 12.4

**Comment by kdaily** [comment]
I reproduced this. Debug logs:

```
(awscli-7199) ➜  awscli-7199 aws configure --profile asdfasdf

The config profile (asdfasdf) could not be found
(awscli-7199) ➜  awscli-7199 aws configure --profile asdfasdf --debug
2022-08-17 14:48:17,025 - MainThread - awscli.clidriver - DEBUG - CLI version: aws-cli/1.25.54 Python/3.10.4 Darwin/21.5.0 botocore/1.27.54
2022-08-17 14:48:17,025 - MainThread - awscli.clidriver - DEBUG - Arguments entered to CLI: ['configure', '--profile', 'asdfasdf', '--debug']
2022-08-17 14:48:17,025 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function add_scalar_parsers at 0x102efb640>
2022-08-17 14:48:17,025 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function register_uri_param_handler at 0x102a19480>
2022-08-17 14:48:17,025 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function inject_assume_role_provider_cache at 0x102a71d80>
2022-08-17 14:48:17,025 - MainThread - awscli.customizations.assumerole - DEBUG - ProfileNotFound caught when trying to inject assume-role cred provider cache.  Not configuring JSONFileCache for assume-role.
2022-08-17 14:48:17,025 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function attach_history_handler at 0x102db6680>
2022-08-17 14:48:17,025 - MainThread - botocore.hooks - DEBUG - Event building-command-table.configure: calling handler <function add_waiters at 0x102f08ca0>
2022-08-17 14:48:17,025 - MainThread - botocore.hooks - DEBUG - Event before-building-argument-table-parser.configure: calling handler <function update_endpoint_url at 0x102f0b250>
2022-08-17 14:48:17,026 - MainThread - awscli.clidriver - DEBUG - Exception caught in main()
Traceback (most recent call last):
  File "/Users/kdaily/.pyenv/versions/3.10.4/envs/awscli-7199/lib/python3.10/site-packages/awscli/clidriver.py", line 217, in main
    return command_table[parsed_args.command](remaining, parsed_args)
  File "/Users/kdaily/.pyenv/versions/3.10.4/envs/awscli-7199/lib/python3.10/site-packages/awscli/customizations/commands.py", line 136, in __call__
    self._session.emit(event, argument_table=self._arg_table, args=args,
  File "/Users/kdaily/.pyenv/versions/3.10.4/envs/awscli-7199/lib/python3.10/site-packages/botocore/session.py", line 763, in emit
    return self._events.emit(event_name, **kwargs)
  File "/Users/kdaily/.pyenv/versions/3.10.4/envs/awscli-7199/lib/python3.10/site-packages/botocore/hooks.py", line 412, in emit
    return self._emitter.emit(aliased_event_name, **kwargs)
  File "/Users/kdaily/.pyenv/versions/3.10.4/envs/awscli-7199/lib/python3.10/site-packages/botocore/hooks.py", line 256, in emit
    return self._emit(event_name, kwargs)
  File "/Users/kdaily/.pyenv/versions/3.10.4/envs/awscli-7199/lib/python3.10/site-packages/botocore/hooks.py", line 239, in _emit
    response = handler(**kwargs)
  File "/Users/kdaily/.pyenv/versions/3.10.4/envs/awscli-7199/lib/python3.10/site-packages/awscli/customizations/overridesslcommonname.py", line 120, in update_endpoint_url
    region = session.get_config_variable("region")
  File "/Users/kdaily/.pyenv/versions/3.10.4/envs/awscli-7199/lib/python3.10/site-packages/botocore/session.py", line 305, in get_config_variable
    return self.get_component('config_store').get_config_variable(
  File "/Users/kdaily/.pyenv/versions/3.10.4/envs/awscli-7199/lib/python3.10/site-packages/botocore/configprovider.py", line 426, in get_config_variable
    return provider.provide()
  File "/Users/kdaily/.pyenv/versions/3.10.4/envs/awscli-7199/lib/python3.10/site-packages/botocore/configprovider.py", line 628, in provide
    value = provider.provide()
  File "/Users/kdaily/.pyenv/versions/3.10.4/envs/awscli-7199/lib/python3.10/site-packages/botocore/configprovider.py", line 718, in provide
    scoped_config = self._session.get_scoped_config()
  File "/Users/kdaily/.pyenv/versions/3.10.4/envs/awscli-7199/lib/python3.10/site-packages/botocore/session.py", line 404, in get_scoped_config
    raise ProfileNotFound(profile=profile_name)
botocore.exceptions.ProfileNotFound: The config profile (asdfasdf) could not be found
2022-08-17 14:48:17,027 - MainThread - awscli.clidriver - DEBUG - Exiting with rc 255

The config profile (asdfasdf) could not be found
```

**Comment by kdaily** [comment]
Thanks again for reporting. We've reverted the changes causing this issue and it should be resolved with version 1.25.55.

**Comment by kbehlers** [comment]
Confirmed, 1.25.55 is fixed 🎉 Thank you!
